custom_test.go

```diff
@@ -0,0 +1,87 @@
+package msgpack_test
+
+import (
+	"bytes"
+	"reflect"
+	"testing"
+
+	"gopkg.in/vmihailenco/msgpack.v2"
+)
+
+// ------------------------------------------------------------------------
+
+func init() {
+	msgpack.Register(reflect.TypeOf(customTestSet{}), encodeCustomTestSet, decodeCustomTestSet)
```

Comment:
BTW please use `msgpack.Marshal`/`msgpack.Unmarshal`. I plan to remove `msgpack.Register` from the future version.